grakn-migration/csv/src/main/java/ai/grakn/migration/csv/Main.java

```diff
@@ -40,22 +40,32 @@
  */
 public class Main {
 
-    static boolean newClusterManager = false;
-
     public static void main(String[] args) {
         start(null, args);
     }
 
     public static void start(TaskManager manager, String[] args){
+        boolean isNewClusterManager = false;
         if(manager == null){
+            isNewClusterManager = true;
             manager = new DistributedTaskManager();
         }
 
-        TaskManager finalManager = manager;
-        MigrationCLI.init(args, CSVMigrationOptions::new).stream()
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .forEach((options) -> runCSV(finalManager, options));
+        try {
+            TaskManager finalManager = manager;
+            MigrationCLI.init(args, CSVMigrationOptions::new).stream()
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .forEach((options) -> runCSV(finalManager, options));
+        } finally {
+            if (isNewClusterManager) {
+                try {
+                    manager.close();
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
We should provide more details in addition to just a stack trace dump. As a user if migration finishes with a stack trace dump I would be extremely confused. 

Did my data go through? Which part exactly failed?

In the above scenario it looks like it is possible for the data to go through but to finish with a stack trace. (If the manager fails to close). Hence the confusion.